core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.kt

```diff
@@ -120,106 +122,142 @@ class STDCMPostProcessing(private val graph: STDCMGraph) {
     }
-    return res
-}
 
-/** Computes the departure time, made of the sum of all delays added over the path */
-fun computeDepartureTime(edges: List<STDCMEdge>, startTime: Double): Double {
-    var mutStartTime = startTime
-    for (edge in edges) mutStartTime += edge.addedDelay
-    return mutStartTime
-}
+    /** Computes the departure time, made of the sum of all delays added over the path */
+    private fun computeDepartureTime(edges: List<STDCMEdge>, startTime: Double): Double {
+        var addedDelay = 0.0
+        for (edge in edges) addedDelay += edge.addedDelay
+        val totalAddedDelay = addDelayForPlannedNodes(edges, addedDelay)
+        return startTime + totalAddedDelay
+    }
 
-/** Builds the list of stops from the edges */
-private fun makeStops(edges: List<STDCMEdge>): List<TrainStop> {
-    val res = ArrayList<TrainStop>()
-    var offset = 0.meters
-    for (edge in edges) {
-        val prevNode = edge.previousNode
-        // Ignore first path node and last node (we aren't checking lastEdge.getEdgeEnd())
-        if (
-            prevNode.previousEdge != null &&
-                prevNode.stopDuration != null &&
-                prevNode.stopDuration >= 0
-        )
-            res.add(
-                TrainStop(
-                    offset.meters,
-                    prevNode.stopDuration,
-                    // TODO: forward and use onStopSignal param from request
-                    isTimeStrictlyPositive(prevNode.stopDuration)
-                )
+    /** Add delay needed to minimize the relative time diff with planned timing data */
+    private fun addDelayForPlannedNodes(edges: List<STDCMEdge>, addedDelay: Double): Double {
+        val nodes = getNodes(edges)
+        val maxNodeDelay = getMaxNodeDelay(nodes)
+        val plannedNodes = getPlannedNodes(nodes)
+        var delayForPlannedNodes = 0.0
+        for (plannedNode in plannedNodes) {
+            val timeDiff = plannedNode.getTimeDiffWithPlannedArrival(addedDelay)
+            // Train passes before planned node: adding delay can make the train pass closer to
+            // planned arrival time
+            if (timeDiff < 0.0) {
+                delayForPlannedNodes = abs(timeDiff)
+                break
+            }
+        }
+        // Total delay added shouldn't be higher than the maximum delay that can possibly be added
+        // to the nodes.
+        return min(addedDelay + delayForPlannedNodes, maxNodeDelay)
+    }
+
+    private fun getNodes(edges: List<STDCMEdge>): List<STDCMNode> {
+        val nodes = edges.map { it.previousNode }.toMutableList()
+        nodes.add(edges.last().getEdgeEnd(graph))
+        return nodes
+    }
+
+    private fun getPlannedNodes(nodes: List<STDCMNode>): List<STDCMNode> {
+        return nodes.filter { it.plannedTimingData != null }
+    }
+
+    private fun getMaxNodeDelay(nodes: List<STDCMNode>): Double {
+        return nodes.minOfOrNull { it.maximumAddedDelay + it.totalPrevAddedDelay }!!
+    }
```

Comment:
That method returns the maximum *total* added delay (compared to the min start time). But it's used to cap the delay *on top* of the one that's already added to avoid conflicts

It should probably just be `nodes.last().maximumAddedDelay` given how it's used

Reply:
This should be correct. This value gives the maximum possible delay that can be added to the start time without encountering any sort of conflict. I compare this value to the total delay added + additional delay i want to add for the planned timing data.

Reply:
Nevermind I misread line 141